crowbar_framework/app/controllers/machines_controller.rb

```diff
@@ -88,10 +88,19 @@ def role
 
   add_help(:rename, [:id])
   def rename
-    @machine.alias = params[:alias]
+    begin
+      @machine.alias = params[:alias]
+    rescue StandardError => e
+      alias_error = e.message
+    end
 
     respond_to do |format|
-      if @machine.save
+      if !alias_error.nil?
```

Comment:
i think that won't work in the case where the variable doesn't exist. you'd have to set it to nil explicitly somewhere above